code/modules/vending/mail.dm

```diff
@@ -0,0 +1,308 @@
+#define STATE_SORTING "sorting"
+#define STATE_IDLE "idle"
+#define STATE_YES "yes"
+#define STATE_NO "no"
+
+/obj/machinery/mailsorter
+	name = "mail sorter"
+	desc = "A large mail sorting unit. Sorting mail since 1987!"
+	icon = 'icons/obj/machines/mailsorter.dmi'
+	icon_state = "mailsorter"
+	layer = BELOW_OBJ_LAYER
+	density = TRUE
+	max_integrity = 300
+	integrity_failure = 0.33
+	var/light_mask = "mailsorter-light-mask"
+	var/panel_type = "panel"
+
+	circuit = /obj/item/circuitboard/machine/mailsorter
+
+	/// How much mail can the mail sorter store.
+	var/mail_capacity = 100
+	/// What the machine is currently doing. Can be "sorting", "idle", "yes", "no".
+	var/currentstate = STATE_IDLE
+	/// List of all mail that's inside the mailbox.
+	var/list/mail_list = list()
+	/// The direction in which the mail will be unloaded.
+	var/output_dir = SOUTH
+	/// List of the departments to sort the mail for.
+	var/list/sorting_departments = list(
+		DEPARTMENT_ENGINEERING,
+		DEPARTMENT_SECURITY,
+		DEPARTMENT_MEDICAL,
+		DEPARTMENT_SCIENCE,
+		DEPARTMENT_CARGO,
+		DEPARTMENT_SERVICE,
+		DEPARTMENT_COMMAND
+	)
+	var/list/choices = list(
+		"Eject" = icon('icons/hud/radial.dmi', "radial_eject"),
+		"Dump" = icon('icons/hud/radial.dmi', "mail_dump"),
+		"Sort" = icon('icons/hud/radial.dmi', "mail_sort")
+	)
+
+	req_access = list(ACCESS_CARGO)
+
+/obj/machinery/mailsorter/proc/get_unload_turf()
+	return get_step(src, output_dir)
+
+/obj/machinery/mailsorter/screwdriver_act(mob/living/user, obj/item/tool)
+	default_deconstruction_screwdriver(user, "mailsorter-off", "mailsorter", tool)
+	update_appearance()
+	return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/mailsorter/crowbar_act(mob/living/user, obj/item/tool)
+	default_deconstruction_crowbar(tool)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/mailsorter/Initialize(mapload)
+	. = ..()
```

Comment:
This isn't doing anything?

Reply:
Exactly. It was used to init the dropping turf, but now it's dynamic to prevent hard deletes

Reply:
So delete it please